engine.py

```diff
@@ -197,6 +211,153 @@ def log_error(self, msg):
         
     
     ##########################################################################################
+    # panel interfaces
+    
+    def _generate_panel_id(self, dialog_name, bundle):
+        """
+        Given a dialog name and a bundle, generate a Nuke panel id.
+        This panel id is used by Nuke to identify and persist the panel.
+        
+        This will return something like 'shotgun_tk_multi_loader2_main'
+        
+        :param dialog_name: An identifier string to identify the dialog to be hosted by the panel
+        :param bundle: The bundle (e.g. app) object to be associated with the panel
+        :returns: a unique identifier string 
+        """
+        panel_id = "shotgun_%s_%s" % (bundle.name, dialog_name)
```

Comment:
Ok, so the new implementation now centralizes this aspect. It's an indirect change after some deliberation - i guess the consequence is that this now goes away both in maya and nuke.